### PR TITLE
[Add] charts

### PR DESCRIPTION
**Pull request by siva2204**
- add candle stick and line graph in company page
- all the price recorded in stockhistory is same so while in development it'll be straight line (need trail testing)
- bar chart is easy to added  (can be done later

*have to check with custom stock history data and in trail test as well*

- x-axis shows up weird numbers for now due to time format (can we remove x-axis ?)
- the complete output of charts is not up to the level as of now (can redo everything in vanilla canvas in flutter if we have time) 

#### screenshots
Screenshot from 2022-02-03 15-55-02

Screenshot from 2022-02-02 17-06-25

### CODE REVIEW

**Review by dhruv1294 — APPROVED** [review]

**Comment by dhruv1294** [comment]
can remove commented code if not being used

**Comment by siva2204** [comment]
> can remove commented code if not being used

Will do with other changes

File: lib/blocs/stock_history/history/stock_history_state.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  final String message;
+
+  const StockHistoryError(this.message);
+}
+
+class StockHistorySuccess extends StockHistoryState {
+  final Map<String, StockHistory> stockHistoryMap;
+
+  const StockHistorySuccess(this.stockHistoryMap);
+}
```
Comment (Sudhindra3000) [inline lib/blocs/stock_history/history/stock_history_state.dart:22]: Override props field for both

File: lib/components/graph/chart_enum.dart [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// TODO bar chart
+enum ChartType { line, candlestick, bar }
```
Comment (Sudhindra3000) [inline lib/components/graph/chart_enum.dart:2]: Rename file to chart_type.dart

File: lib/components/graph/stock_chart.dart [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+    // fetching stockHistory
+    context
+        .read<StockHistoryCubit>()
+        .getStockHistory(widget.stockId, currentResolution);
+
+    // subscribing to stockHistory stream
+    context
+        .read<StockHistoryStreamCubit>()
+        .getStockHistoryUpdates(widget.stockId);
```
Comment (Sudhindra3000) [inline lib/components/graph/stock_chart.dart:60]: Subscribe to stream after initial data is fetched?

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

File: lib/components/graph/stock_chart.dart (hunk 3 above)
Comment (siva2204) [reply lib/components/graph/stock_chart.dart:60]: yes, in initial data of 60 `stockHistory` will be returned, the stream will have a single `stockHistory`, which will be added to the graph which is built from the data fetched initially.

File: lib/components/graph/chart_enum.dart (hunk 2 above)
Comment (siva2204) [reply lib/components/graph/chart_enum.dart:2]: okay sur :(

File: lib/components/graph/stock_chart.dart (hunk 3 above)
Comment (Sudhindra3000) [reply lib/components/graph/stock_chart.dart:60]: I meant wait till data is fetched and then subscribe to the stream

Comment (siva2204) [reply lib/components/graph/stock_chart.dart:60]: okay

**Review by Sudhindra3000 — APPROVED** [review]

**Comment by siva2204** [comment]
i'm going ahead to merge this pr, even though there is an [has](https://github.com/fluttercandies/flutter-interactive-chart/issues/14) issue with candle stick, which will not be a issue in real event(removing max price and min price resolves this)

**Pull request merged**